### PR TITLE
UX: Validate embeddings settings

### PR DESCRIPTION
**Pull request by romanrizzi**

### CODE REVIEW

File: lib/embeddings/vector_representations/all_mpnet_base_v2.rb [hunk 1]
```diff
@@ -4,19 +4,37 @@ module DiscourseAi
   module Embeddings
     module VectorRepresentations
       class AllMpnetBaseV2 < Base
+        class << self
+          def name
+            "all-mpnet-base-v2"
+          end
+
+          def correctly_configured?
+            SiteSetting.ai_embeddings_discourse_service_api_key.present? &&
```
Comment (xfalcox) [inline lib/embeddings/vector_representations/all_mpnet_base_v2.rb:13]: API key here is optional

Comment (romanrizzi) [reply lib/embeddings/vector_representations/all_mpnet_base_v2.rb:13]: There are other places where we check for this. Will remove those too

File: lib/embeddings/vector_representations/bge_large_en.rb [hunk 2]
```diff
@@ -4,6 +4,35 @@ module DiscourseAi
...
+        class << self
+          def name
+            "bge-large-en"
+          end
+
+          def correctly_configured?
+            SiteSetting.ai_cloudflare_workers_api_token.present? ||
+              DiscourseAi::Inference::HuggingFaceTextEmbeddings.configured? ||
+              (
+                SiteSetting.ai_embeddings_discourse_service_api_key.present? &&
```
Comment (xfalcox) [inline lib/embeddings/vector_representations/bge_large_en.rb:16]: Ditto

File: lib/embeddings/vector_representations/multilingual_e5_large.rb [hunk 3]
```diff
@@ -4,14 +4,41 @@ module DiscourseAi
...
       class MultilingualE5Large < Base
+        class << self
+          def name
+            "multilingual-e5-large"
+          end
+
+          def correctly_configured?
+            DiscourseAi::Inference::HuggingFaceTextEmbeddings.configured? ||
+              (
+                SiteSetting.ai_embeddings_discourse_service_api_key.present? &&
```
Comment (xfalcox) [inline lib/embeddings/vector_representations/multilingual_e5_large.rb:15]: Ditto

**Review by xfalcox — APPROVED** [review]

**Pull request merged**